quickwit-indexing/src/actors/doc_processor.rs

```diff
@@ -0,0 +1,439 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use quickwit_actors::{Actor, ActorContext, ActorExitStatus, Handler, Mailbox, QueueCapacity};
+use quickwit_common::runtimes::RuntimeType;
+use quickwit_doc_mapper::{DocMapper, DocParsingError};
+use tantivy::schema::{Field, Value};
+use tokio::runtime::Handle;
+use tracing::warn;
+
+use crate::actors::Indexer;
+use crate::models::{NewPublishLock, PreparedDoc, PreparedDocBatch, PublishLock, RawDocBatch};
+
+enum PrepareDocumentError {
+    ParsingError,
+    MissingField,
+}
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct DocProcessorCounters {
+    /// Overall number of documents received, partitioned
+    /// into 3 categories:
+    /// - number docs that did not parse correctly.
+    /// - number docs missing a timestamp (if the index has no timestamp,
+    /// then this counter is 0)
+    /// - number of valid docs.
+    pub num_parse_errors: u64,
+    pub num_missing_fields: u64,
```

Comment:
`num_missing_fields` is a bit misleading. Maybe `num_docs_with_missing_fields`

Reply:
There is also the namings of the `INDEXER_METRICS`:
`parsing_errors_num_docs_total`, `missing_field_num_docs_total`, `missing_field_num_bytes_total``

We also use both `overall` and `total`.

A solution could be to align names with the `INDEXER_METRICS` for more clarity.

(This can be handled in another PR done by someoneelse).

Reply:
total is something imposed by prometheus naming convention.